mantidproject/mantid
# Peak range property changed to take spectra

**Description of work.**
The property `PeakRange` in `ReflectometryBackgroundSubtraction` has been changed so that it can take either spectrum number or workspace index depending on the value of  `InputWorkspaceIndexType`. 
The property `InputWorkspaceIndexSet` has also be renamed to `ProcessingInstructions` so it consistent with other ISIS reflectometry algorithms.

**To test:**
Run the following script, it will check that using the spectrum numbers and the workspace index give the same output.
```
ws = LoadISISNexus("offspec49949", LoadMonitors = "Exclude")

bkg_subtr_index = ReflectometryBackgroundSubtraction(ws, InputWorkspaceIndexType = "WorkspaceIndex", ProcessingInstructions = "200-600", BackgroundCalculationMethod = "AveragePixelFit", PeakRange ="350-450")
bkg_subtr_spectra = ReflectometryBackgroundSubtraction(ws, InputWorkspaceIndexType = "SpectrumNumber", ProcessingInstructions ="204-604", BackgroundCalculationMethod = "AveragePixelFit", PeakRange ="354-454")

CompareWorkspaces(bkg_subtr_index , bkg_subtr_spectra)
```
Fixes #25822 

*This does not require release notes* because the `AveragePixelFit` method in this algorithm has been added during this release period

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
